### PR TITLE
Dockerized project

### PR DESCRIPTION
**Pull request by procrastinatio**
Very basic, but running (only PG-based services) version of `mf-chsdi3`

Legacy `python2`only version should still work and deploy: http://mf-chsdi3.dev.bgdi.ch/feature_BGDIINF_SB_1899/

**Codebuild**
On python2, tests are only tests... It doesn't mean the project may deploy and/or even work in a `vhost`
Python3 and docker: tests are passing, docker image is build and runs, and basic service is responding...

See https://jira.swisstopo.ch/browse/BGDIINF_SB-1894 for more

### CODE REVIEW

**Comment by hansmannj** [comment]
Merci @procrastinatio!
Good to have you on board with this :slightly_smiling_face:

**Comment by procrastinatio** [comment]
Why is it not building?

**Comment by procrastinatio** [comment]
Passing :smile:

**Comment by procrastinatio** [comment]
Rebased.
@hansmannj  Try to emphase on files which are in both side (python2,3), like `Makefile`, `setup.py` and `apache/wsgi.conf.in`. I have to check again, but I think the former file is only for recent apache version and may not work with current vhost.

File: dev.env [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+DEBUG=True
+APACHE_ENTRY_PATH=
+APACHE_BASE_PATH=main
+BRANCH_STAGING=
+GIT_BRANCH=
+MODWSGI_USER=www-data
+INSTALLDIR=/var/www/vhosts/mf-chsdi3/private/chsdi
+WSGI_APP=/var/www/vhosts/mf-chsdi3/private/chsdi/apache/application.wsgi
+KML_TEMP_DIR=/var/local/print/kml
+DEPLOY_TARGET=prod
```
Comment (hansmannj) [inline dev.env:20]: Is the DEPLOY_TARGET=prod correct here, or should it be dev?

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,8 @@
 SHELL = /bin/bash
```
Comment (hansmannj) [inline Makefile:1]: This file is a monster of a Makefile

File: dev.env [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+MODWSGI_CONFIG=production.ini
```
Comment (hansmannj) [inline dev.env:21]: is production.ini correct in dev.env?

File: dev.env [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ROBOTS_FILE=robots_prod.txt
```
Comment (hansmannj) [inline dev.env:22]: is there a robots_dev.txt?

File: int.env [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+DBPORT=5432
+DBSTAGING=int
+DEPLOY_TARGET=int
+GEODATA_STAGING=integration
+GEOADMIN_FILE_STORAGE_BUCKET=public-int-bgdi-ch
+GEOADMINHOST=mf-geoadmin3.int.bgdi.ch
+HOST=mf-chsdi3.int.bgdi.ch
+KEEP_VERSION=true
+WMTS_PUBLIC_HOST=tod.int.bgdi.ch
+MODWSGI_CONFIG=production.ini
```
Comment (hansmannj) [inline int.env:15]: Is there a integration.ini instead?

**Comment by hansmannj** [comment]
Thanks @procrastinatio for digging in the mud. This looks promising! :-)

File: Jenkinsfile [hunk 6]
```diff
@@ -2,6 +2,10 @@
```
[the remark this answers is not in the document]
Comment (procrastinatio) [reply Jenkinsfile:2]: Indeed

File: buildspec.yml [hunk 7]
```diff
@@ -36,16 +36,19 @@ env:
     CACHE_CONTROL: no-cache
```
[the remark this answers is not in the document]
Comment (procrastinatio) [reply buildspec.yml:36]: Yes. But I really don't want to rename anything that is needed by the `vhost`. You cannot imagine how complexe it is.

File: buildspec2.yml [hunk 8]
```diff
@@ -0,0 +1,126 @@
+#
```
[the remark this answers is not in the document]
Comment (procrastinatio) [reply buildspec2.yml:1]: Yes

**Comment by procrastinatio** [comment]
@hansmannj  thank you for the review. Now I have to make the whole thing work again in a `vhost` :-(

File: dev.env (hunk 4 above)
Comment (procrastinatio) [reply dev.env:22]: Well, _jein_, it's actually called `robots.txt` (don't ask why)

File: int.env (hunk 5 above)
Comment (procrastinatio) [reply int.env:15]: No, on `prod` and `int`  the `production.ini` is used. On `dev` it's `development.ini` (which sources `production.ini`). Again, don't ask why

File: Makefile [hunk 9]
```diff
@@ -220,34 +249,62 @@ help:
...
+templates: apache/wsgi.conf apache/application.wsgi development.ini production.ini chsdi/static/info.json
+	$(call build_templates,$(DEPLOY_TARGET))
+
+
+.PHONY: baseimage
+baseimage:
+	docker build -t $(DOCKER_IMAGE_LOCAL_TAG_BASEIMAGE)  -f Dockerfile.base .
 
 
+.PHONY: image
```
[the remark this answers is not in the document]
Comment (procrastinatio) [reply Makefile:274]: OK

**Comment by procrastinatio** [comment]
Basically, it also build and deploys in the legacy `vhosts`

File: Dockerfile.base [hunk 10]
```diff
@@ -0,0 +1,14 @@
+FROM python:3.7-buster
```
[the remark this answers is not in the document]
Comment (procrastinatio) [reply Dockerfile.base]: OK, deleted

File: Makefile [hunk 11]
```diff
@@ -220,34 +249,62 @@ help:
...
 all: setup chsdi/static/css/extended.min.css templates translate lint fixrights doc rss
 
 setup: .venv node_modules .venv/hooks
 
-templates: apache/wsgi.conf development.ini production.ini chsdi/static/info.json
+templates: apache/wsgi.conf apache/application.wsgi development.ini production.ini chsdi/static/info.json
+	$(call build_templates,$(DEPLOY_TARGET))
+
+
+.PHONY: baseimage
```
[the remark this answers is not in the document]
Comment (procrastinatio) [reply Makefile]: Right

File: Makefile (hunk 2 above)
Comment (procrastinatio) [reply Makefile:1]: Really?

File: dev.env (hunk 3 above)
Comment (procrastinatio) [reply dev.env:21]: Probably no. But these variables are only relevant at built time. It should be improved later.

File: Dockerfile [hunk 12]
```diff
@@ -0,0 +1,46 @@
+FROM python:3.7-buster
```
[the remark this answers is not in the document]
Comment (procrastinatio) [reply Dockerfile:1]: I did remove it

File: dev.env (hunk 1 above)
Comment (procrastinatio) [reply dev.env:20]: No, but for now it's only relevant at built time.

**Review by hansmannj — APPROVED** [review]
:+1: Looks (almost) ;-) good now, thanks a lot!

As discussed, some things for a new PR would be to clean-up and change the way of using env vars in buildspec files, dockerfiles, Makefile et al.

Also the new Makefile targets are not yet in the help section of the Makefile. So in a way they are "un-documented" currently and should be added.

Might be good to merge this PR, so it doesn't grow too big and continue in a new PR.

File: apache/ports.conf.in [hunk 13]
```diff
@@ -0,0 +1,9 @@
+Listen ${APACHE_PORT}
+
+<IfModule mod_ssl.c>
+    Listen 443
```
Comment (boecklic) [inline apache/ports.conf.in:4]: SSL offloading for services running on k8s will be done on cloudfront level, so service should listen on port 80

**Pull request merged**